server/data/data-library/src/main/java/org/apache/james/dlp/eventsourcing/EventSourcingDLPConfigurationStore.java

```diff
@@ -78,4 +80,11 @@ public void clear(Domain domain) {
         eventSourcingSystem.dispatch(new ClearCommand(domain));
     }
 
+    @Override
+    public Optional<DLPConfigurationItem> fetch(Domain domain, Id ruleId) {
```

Comment:
well, it could by the default implementation in DLPConfigurationStore but it looks dangerous to provide a method that will load an aggregate from history, compute a list than filter a given rule. I'm pretty sure people will then call each rule after each other.

What about computing a projection ? What is the use case of this API ?

Reply:
why does it look dangerous?

Reply:
dangerous as in "very slow" and "cpu intensive"

Reply:
Well not as "slow" and "cpu intensive" than having the client fetch all rules, then apply filtering on the ID on his side.

BTW if it turns out to be too slow we can improve this with a projection.

Reply:
ok we decided to let it like that for the moment, and to improve it if we have some performance issues later